Add possibility for legacy password upgrade (#7745)
sulu/sulu: UPGRADE.md

# Upgrade

## 2.5.23

### User entity method return types changed

> The sulu `User` entity requires the following changes:

Reviewer: ```suggestion
## 2.5.23

### User getSalt method return types changed

To support upgrade of `Legacy` passwords hashes from Sulu 1.6, the Sulu `User::getSalt` method requires the following changes if you have overwritten it:
```

```diff
-public function getSalt();
+public function getSalt(): ?string;
```

> 

Reviewer: ````suggestion

If migrating from an old Sulu 1.6 project you can configure a legacy hasher to seamlessly upgrade the Users password:

<details>
<summary>Example Password Upgrade configuration:</summary>

```yaml
# config/packages/security.yaml
security:
    # ...
    password_hashers:
        legacy:
            algorithm: sha512
            iterations: 5000
            encode_as_base64: false

        Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface:
            algorithm: bcrypt
            migrate_from:
                - legacy
    # ...
```

See also the Symfony Password Upgrade Documentation [here](https://symfony.com/doc/6.4/security/passwords.html#upgrade-the-password).

</details>

````

# Upgrade

## 2.5.23

### User getSalt method return types changed

> To support upgrade of `Legacy` passwords hashes from Sulu 1.6, the Sulu `User::getSalt` method requires the following changes if you have overwritten it:

Reviewer 2: ```suggestion
To support the upgrade of `Legacy` password hashes from Sulu 1.6, the `User::getSalt` method requires the following changes if you have overwritten it:
```

```diff
-public function getSalt();
+public function getSalt(): ?string;
```

> If migrating from an old Sulu 1.6 project you can configure a legacy hasher to seamlessly upgrade the Users password:

Reviewer 2: ```suggestion
If migrating from an old Sulu 1.6 project, you can configure a legacy hasher to seamlessly upgrade the user's password:
```

> <details>

Reviewer: did put this in to a `details` tag as it is not a required change.
